themes/powerbash10k/powerbash10k.theme.sh

```diff
@@ -0,0 +1,367 @@
+    box="[|]"
+    info="t:$(todo.sh ls | command grep -E "TODO: [0-9]+ of ([0-9]+)" | awk '{ print $4 }' )"
+    printf "%s|%s|%s|%s" "$color" "$info" "$_omb_prompt_bold_green" "$box"
+}
+
+function __pb10k_prompt_clock {
+    local color box info
+    [ "$THEME_SHOW_CLOCK" != "true" ] && return
+    color=$THEME_CLOCK_COLOR
+    box=""
+    info=" $(date +"$THEME_CLOCK_FORMAT")"
+    printf "%s|%s|%s|%s" "$color" "$info" "$_omb_prompt_bold_purple" "$box"
+}
+
+function __pb10k_prompt_battery {
+    local color box info
+    [ ! -e "$OSH/plugins/battery/battery.plugin.sh" ] ||
+    [ "$THEME_SHOW_BATTERY" != "true" ] && return
+    info=$(battery_percentage)
+    color=$_omb_prompt_bold_green
+    if [ "$info" -lt 50 ]; then
+        color=$_omb_prompt_bold_olive
+    elif [ "$info" -lt 25 ]; then
+        color=$_omb_prompt_bold_brown
+    fi
+    box="[|]"
+    ac_adapter_connected && info+="+"
+    [ "$info" == "100+" ] && info="AC"
+    printf "%s|%s|%s|%s" "$color" "$info" "$_omb_prompt_bold_white" "$box"
+}
+
+function __pb10k_prompt_exitcode {
+    local color
+    [ "$THEME_SHOW_EXITCODE" != "true" ] && return
+    color=$_omb_prompt_bold_red
+    [ "$exitcode" -ne 0 ] && printf "%s|%s" "$color" "✘ ${exitcode}"
+}
+
+function __pb10k_prompt_char {
+    local color
+    color=$(if [ "$exitcode" -ne 0 ]; then
+            echo "$_omb_prompt_bold_red"
+        else
+            echo "$_omb_prompt_bold_green"
+    fi)
+    printf "%s|%s" "$color" "$__PB10K_PROMPT_CHAR_PS1"
+}
+
+function __pb10k_prompt_cmd_duration {
+    local color box info
+    color=$_omb_prompt_bold_navy
+    box=""
+    info="$(if [ "$__pb10k_timer_show" -gt 0 ]; then
+        printf " %s" "$(__pb10k_format_duration "$__pb10k_timer_show")"
+    fi)"
+    [ "$info" == "" ] && return
+    printf "%s|%s|%s|%s" "$color" "$info" "$_omb_prompt_bold_white" "$box"
+}
+
+# cli
+function __pb10k_show {
+    local seg
+    seg="${1:-}"
+    shift
+    export THEME_SHOW_"$seg"=true
+}
+
+function __pb10k_hide {
+    local seg
+    seg="${1:-}"
+    shift
+    export THEME_SHOW_"$seg"=false
+}
+
+function __pb10k_completion {
+    local cur action actions segments
+    COMPREPLY=()
+    cur="${COMP_WORDS[COMP_CWORD]}"
+    action="${COMP_WORDS[1]}"
+    actions="show hide"
+    segments="battery clock exitcode python ruby scm sudo todo"
+    case "$action" in
+        show)
+        COMPREPLY=( "$(compgen -W "$segments" -- "$cur")" )
+        return 0
+        ;;
+        hide)
+        COMPREPLY=( "$(compgen -W "$segments" -- "$cur")" )
+        return 0
+        ;;
+    esac
+
+    COMPREPLY=( "$(compgen -W "$actions" -- "$cur")" )
+    return 0
+}
+
+function pb10k {
+    local action segments
+    action="${1:-}"
+    shift
+    segments="${*:-}"
+    typeset func
+    case "$action" in
+        show)
+        func=__pb10k_show;;
+        hide)
+        func=__pb10k_hide;;
+    esac
+    IFS=" " read -ra segments <<< "$segments"
+    for seg in "${segments[@]}"; do
```

Comment:
```suggestion
    local seg
    for seg in "${segments[@]}"; do
```

Reply:
Sure, I'll localize all these variables.

Reply:
Fixed in https://github.com/ohmybash/oh-my-bash/pull/474/commits/043379ae88aabd7d636999feaf454b289abcc2e5.